packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/utils.js

```diff
@@ -394,3 +394,36 @@ export function build_expression(context, expression, metadata, state = context.
 
 	return sequence;
 }
+
+/**
+ * Wraps a statement/expression with dev stack tracking in dev mode
+ * @param {CallExpression} call_expression - The function call to wrap (e.g., $.if, $.each, etc.)
+ * @param {{ start?: number }} node - AST node for location info
+ * @param {'component' | 'if' | 'each' | 'await' | 'key'} type - Type of block/component
+ * @param {Record<string, number | string>} [additional] - Any additional properties to add to the dev stack entry
+ * @returns {Statement} - Statement with or without dev stack wrapping
+ */
+export function with_dev_stack(call_expression, node, type, additional) {
+	if (!dev) {
+		return b.stmt(call_expression);
+	}
+
+	const location = node.start && locator(node.start);
+	if (!location) {
+		return b.stmt(call_expression);
+	}
+
+	return b.stmt(
+		b.call(
+			'$.with_dev_stack',
+			b.arrow([], b.block([b.stmt(call_expression)])),
+			b.literal(type),
+			b.literal(filename),
+			b.literal(location.line),
+			b.literal(location.column),
+			additional
+				? b.object(Object.entries(additional).map(([key, value]) => b.init(key, b.literal(value))))
+				: b.null
```

Comment:
we can use `&&` here because trailing undefined arguments to `b.call` are ignored

```suggestion
			additional && b.object(Object.entries(additional).map(([k, v]) => b.init(k, b.literal(v))))
```